backend/de.metas.cucumber/src/test/java/de/metas/cucumber/stepdefs/shipment/pickingterminal/Picking_Terminal_StepDef.java

```diff
@@ -96,4 +112,118 @@ public void pickQty(@NonNull final DataTable dataTable)
+	}
+
+	@And("process picking")
+	public void process_picking(@NonNull final DataTable dataTable)
+	{
+		final List<Map<String, String>> rows = dataTable.asMaps();
+		for (final Map<String, String> row : rows)
+		{
+			final String huIdentifier = DataTableUtil.extractStringForColumnName(row, COLUMNNAME_M_HU_ID + "." + TABLECOLUMN_IDENTIFIER);
+			final String shipmentScheduleIdentifier = DataTableUtil.extractStringForColumnName(row, COLUMNNAME_M_ShipmentSchedule_ID + "." + TABLECOLUMN_IDENTIFIER);
+
+			final I_M_HU hu = huTable.get(huIdentifier);
+			assertThat(hu).isNotNull();
+
+			final I_M_ShipmentSchedule shipmentSchedule = shipmentScheduleTable.get(shipmentScheduleIdentifier);
+			assertThat(shipmentSchedule).isNotNull();
+
+			final String errorMessage = DataTableUtil.extractStringOrNullForColumnName(row, "OPT.ErrorMessage");
+
+			try
+			{
+				pickingCandidateService.processForHUIds(ImmutableSet.of(HuId.ofRepoId(hu.getM_HU_ID())), ShipmentScheduleId.ofRepoId(shipmentSchedule.getM_ShipmentSchedule_ID()));
+
+				Assertions.assertThat(errorMessage).as("ErrorMessage should be null if pickingCandidateService.processForHUIds() finished with no error!").isNull();
+			}
+			catch (final Exception e)
+			{
+				if (Check.isNotBlank(errorMessage))
+				{
+					assertThat(e.getMessage()).contains(errorMessage);
+				}
+				else
+				{
+					throw e;
+				}
+			}
+		}
+	}
+
+	@And("^validate M_HUs are available to pick for shipmentSchedule identified by (.*)$")
+	public void validate_M_HUs_available_to_pick_for_SS(@NonNull final String shipmentScheduleIdentifier, @NonNull final DataTable table)
+	{
+		final I_M_ShipmentSchedule shipmentSchedule = shipmentScheduleTable.get(shipmentScheduleIdentifier);
+		assertThat(shipmentSchedule).isNotNull();
+
+		final IHUPickingSlotBL.PickingHUsQuery query = IHUPickingSlotBL.PickingHUsQuery.builder()
+				.onlyTopLevelHUs(false)
+				.onlyIfAttributesMatchWithShipmentSchedules(true)
+				.shipmentSchedule(shipmentSchedule)
+				.build();
+
+		final List<I_M_HU> availableHUsToPick = huPickingSlotBL.retrieveAvailableHUsToPick(query);
+		assertThat(availableHUsToPick).isNotNull();
+
+		final List<Map<String, String>> rows = table.asMaps();
+		assertThat(availableHUsToPick.size()).isEqualTo(rows.size());
+
+		for (int huIndex = 0; huIndex < availableHUsToPick.size(); huIndex++)
+		{
+			final I_M_HU actualHU = availableHUsToPick.get(huIndex);
+			final Map<String, String> expectedHU = rows.get(huIndex);
+
+			final String huIdentifier = DataTableUtil.extractStringForColumnName(expectedHU, I_M_HU.COLUMNNAME_M_HU_ID + "." + TABLECOLUMN_IDENTIFIER);
+			final I_M_HU hu = huTable.get(huIdentifier);
+			assertThat(hu).isNotNull();
+
+			assertThat(actualHU.getM_HU_ID()).isEqualTo(hu.getM_HU_ID());
+		}
+	}
+
+	@And("^validate that there are no M_HUs available to pick for shipmentSchedule identified by (.*)$")
+	public void validate_no_M_HUs_available_to_pick_for_SS(@NonNull final String shipmentScheduleIdentifier)
```

Comment:
```suggestion
	public void validate_no_M_HUs_available_to_pick_for_ShipmentSched(@NonNull final String shipmentScheduleIdentifier)
```

..call it a german spleen if you want..